### PR TITLE
React widget Blog Post fixes

### PR DESCRIPTION
**Pull request by leebrandt**
## Description:
- Changed widget install to 2.3.0
- Removed `npm install` from beginning of article
- Added instructions for fixin NPM dependency issues in widget
- Copied over @nbarbettini changes from PR #1389

### CODE REVIEW

File: _source/_posts/2017-03-30-react-okta-sign-in-widget.md [hunk 1]
```diff
@@ -14,39 +14,37 @@ Start by cloning the simple React seed project.
 ```bash
 git clone https://github.com/leebrandt/simple-react-seed.git okta-react-widget-sample
 cd okta-react-widget-sample
-npm install
-npm start
```
Comment (mraible) [inline _source/_posts/2017-03-30-react-okta-sign-in-widget.md]: Why did you remove these instructions to install and start?

File: _source/_posts/2017-03-30-react-okta-sign-in-widget.md [hunk 2]
```diff
@@ -252,20 +261,43 @@ export default class LoginPage extends React.Component{
...
+
+Add a Login link to the top navbar by editing `./src/components/common/Navigation.js` and adding a new link under the existing Contact link:
+
+```html
+<li><Link to="login">Login</Link></li>
+```
+
 ## Check It Out
+
+Now install the NPM packages:
```
Comment (mraible) [inline _source/_posts/2017-03-30-react-okta-sign-in-widget.md]: Minor nit - I've always seen npm spelled in lowercase, never uppercase.

File: _source/_posts/2017-03-30-react-okta-sign-in-widget.md (hunk 1 above)
Comment (leebrandt) [reply _source/_posts/2017-03-30-react-okta-sign-in-widget.md]: Widget dependency tree has `q` 1.4.1, but if you `npm install` early there is a dependency on `q` 1.5.1 that gets installed, so when you add the widget it doesn't get downgraded and the widget just spins. work around for widget dependency issues.

Comment (jmelberg-okta) [reply _source/_posts/2017-03-30-react-okta-sign-in-widget.md]: @leebrandt You can get around this by importing the widget via `/dist/js/okta-signin-widget.min.js` instead of the module.

Curious what you think is the better developer experience?

Comment (leebrandt) [reply _source/_posts/2017-03-30-react-okta-sign-in-widget.md]: It should work either way. I don't know that one is better than the other, but not sure why a minor version release caused the widget to not work. Sounds like either we're doing something strange with promises in q (don't see what we COULD be doing), or the q library is sending breaking changes in minor versions (more likely).

Comment (leebrandt) [reply _source/_posts/2017-03-30-react-okta-sign-in-widget.md]: I think from a React standpoint, React developers are used to using the modules. Webpack actually follows the module imports in order to determine dependencies and try to make the final minified file as small as possible (AFAIK). Not sure how/if that will be affected by CDN or script import.

Comment (jmelberg-okta) [reply _source/_posts/2017-03-30-react-okta-sign-in-widget.md]: For context: This is more of an npm problem since `1.11.` (I believe). `okta-auth-js` uses a specific version of `q`, which could result in multiple versions of `q` installed in `node_modules`. This issue only seems to arise when it is imported in as a module.

Our team is looking at a fix for this soon -- but this in the meantime the workaround is to import in the minified `js` instead.
